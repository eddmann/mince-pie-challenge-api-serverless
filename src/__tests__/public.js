import handler from '../handlers/public';

it('successfully responds with no user id', async () => {
  const services = {};

  const response = parseResponse(await handler(services)({}, {}));

  expect(response.statusCode).toBe(200);
  expect(response.body.userId).toBe('N/A');
});

it('successfully responds with no user id when access token found', async () => {
  const services = { getUserIdFromToken: () => Promise.resolve('USER_ID') };

  const response = parseResponse(await handler(services)({ headers: { Authorization: 'TOKEN' } }, {}));

  expect(response.statusCode).toBe(200);
  expect(response.body.userId).toBe('N/A');
});
