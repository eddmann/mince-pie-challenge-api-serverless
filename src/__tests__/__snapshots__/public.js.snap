// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`successfully responds with no user id 1`] = `
Object {
  "body": "{\\"userId\\":\\"N/A\\"}",
  "statusCode": 200,
}
`;

exports[`successfully responds with no user id when access token found 1`] = `
Object {
  "body": "{\\"userId\\":\\"N/A\\"}",
  "statusCode": 200,
}
`;
